### ISSUE TITLE
testport -I -k still kills jails if they fail early

### ISSUE THREAD

**Issue by agrajag9** [freebsd/poudriere#905]
## Prerequisites
- [x] Have you checked for an existing issue describing your problem?
- [x] Are you running the latest version?
- [x] Is your ports tree recent?
- [ ] Is your FreeBSD Host on a [supported release](https://www.freebsd.org/security/#sup)?

## Describe the bug
When `poudriere testport -I -k` encounters a failure during pkg-depends, it still kills jails, rather than leaving them up like described in the man. 

## How to reproduce
1. `poudriere testport -I -k -j somejail -p sometree ports-mgmt/poudriere`
2. https://github.com/freebsd/pkg/issues/1965
3. 
```
[00:00:16] Error: Depends failed to build
[00:00:16] Failed ports: ports-mgmt/dialog4ports:pkg-depends misc/freebsd-release-manifests:pkg-depends lang/perl5.32:pkg-depe
nds
[00:00:16] Skipped ports: security/ca_root_nss
[00:00:16] Cleaning up
[00:00:16] Unmounting file systems
```

## Expected behavior
The jails should remain running so that `pkg.core` can be recovered.

## Screenshots
NA

## Environment
 - Host OS [e.g. 12.2 amd64]: 1400030
 - Jail OS [e.g. 12.0 powerpc]: 1400030
 - Browser: [e.g. chrome, safari]: NA
 - Poudriere Version [e.g. 3.3.1 or git hash or port version]: 3.3.6
 - Ports branch and revision [e.g. 2020Q3 r550754]: fd649cf02561

**Comment by bdrewery** [comment]
In your example testport is for ports-mgmt/poudriere not for its dependencies.  This is working as intended.

**Comment by agrajag9** [comment]
No, this is not the expected behavior. This is a failure, and referencing the man page:
```
     -k		  Do not consider failures as fatal.  Find all failures.
```
This should not have been considered fatal and based on this, with -i or -I the jail should remain running.

If this isn't a bug in poudriere, then the man deserves an explanation.

**Comment by bdrewery** [comment]
Again `-k` is for the port being tested.
The top line of the manpage is **poudriere testport – test a given port's build**. Everything applies to that statement unless otherwise specified. Nothing about `testport` is testing dependencies. It's been like this since 2010.
